src/Sylius/Bundle/RbacBundle/Security/Role/Provider/FallbackHierarchyProvider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\RbacBundle\Security\Role\Provider;
+
+/**
+ * @author Christian Daguerre <christian@daguer.re>
+ */
+class FallbackHierarchyProvider implements HierarchyProviderInterface
+{
+    /**
+     * @var HierarchyProviderInterface
+     */
+    protected $default;
+
+    /**
+     * @var HierarchyProviderInterface
+     */
+    protected $fallback;
+
+    /**
+     * Constructor.
+     *
+     * @param HierarchyProviderInterface $default
+     * @param HierarchyProviderInterface $fallback
+     */
+    public function __construct(HierarchyProviderInterface $default, HierarchyProviderInterface $fallback)
```

Comment:
`$default` -> `$defaultProvider`, `$defaultHierarchyProvider`? Same for `$fallback`.